packages/datadog-plugin-aws-sdk/src/base.js

```diff
@@ -0,0 +1,91 @@
+'use strict'
+
+const Tags = require('opentracing').Tags
+const analyticsSampler = require('../../dd-trace/src/analytics_sampler')
+const awsHelpers = require('./aws_helpers')
+
+function createWrapRequest (tracer, config) {
+  config = normalizeConfig(config)
+
+  return function wrapRequest (request) {
+    return function requestWithTrace (operation, params, cb) {
+      const serviceName = awsHelpers.normalizeServiceName(this)
+
+      const baseTags = {
+        [Tags.SPAN_KIND]: 'client',
+        'span.type': 'http',
+        'service.name': config.service || `${tracer._service}-aws`,
+        'aws.agent': 'js-aws-sdk',
+        'aws.operation': operation,
+        'aws.region': (request.httpRequest && request.httpRequest.region) || this.config.region,
+        'aws.service': serviceName,
+        'component': serviceName
+      }
+
+      const childOf = tracer.scope().active()
+      const span = tracer.startSpan('aws.http', {
+        childOf,
+        tags: baseTags
+      })
+
+      awsHelpers.addAwsRequestTags(span, request, operation, params, serviceName)
+
+      analyticsSampler.sample(span, config.analytics)
+
+      // https://github.com/awsdocs/aws-javascript-developer-guide-v2/blob/
+      // master/doc_source/using-a-callback-function.md#using-an-anonymous-callback-function
+      if (typeof cb === 'function') {
+        return tracer.scope().activate(span, () => {
+          return request.call(this, operation, params, awsHelpers.wrapCallback(tracer, span, cb, childOf, config))
+        })
+      } else {
+        const awsReq = request.apply(this, arguments)
+        const boundAwsReq = tracer.scope().bind(awsReq, span)
+
+        boundAwsReq.on('send', response => {
+          tracer.scope().activate(span)
```

Comment:
This is not doing anything since a callback is not passed.

Reply:
good catch, removed unneeded argument for this listener. I also moved this listener to an earlier event `validate`, adding a comment in a moment discussing this